src/include/cpt/Optional.hpp

```diff
@@ -0,0 +1,41 @@
+//
+// Purpur Tentakel
+// cpp 23
+// 12.05.24
+//
+
+#pragma once
+#include <functional>
+#include <stdexcept>
+#include <variant>
+
+namespace cpt {
+    template<typename T>
+    class Otional final {
+    private:
+        std::variant<T, std::monostate> m_value;
+
+    public:
+        Otional(T value) : m_value{ std::in_place_index<0>, std::forward<T>(value) } { }
+        Otional() : m_value{ std::in_place_index<1>, std::forward<std::monostate>(std::monostate()) } { }
+
+        [[nodiscard]] bool ok() const {
+            return m_value.index() == 0;
+        }
+
+        [[nodiscard]] T unwrap() const {
+            if (ok()) {
+                return std::get<0>(m_value);
+            }
+            throw std::logic_error("while unwrapping an optional");
+        }
```

Comment:
This will only work if `T` is copyable and it will always return a copy. Maybe it should return a `const&` and there should be a second overload that only returns a `&`.

Reply:
do i have to repeat that for `unwrap_or_else` and `unwrap_or_default`
but in that case it is a little bit tricky because `unwrap_or_default` and also `unwrap_or_else` could return a ref tu a temporary then. right?

Reply:
Yes, you would have to repeat that. But you could restrict those functions using `requires`-clauses and make them behave differently based on the value category.

Reply:
i tried.
maybe it worked. 😄